src/Cli/dotnet/commands/dotnet-new/MSBuildEvaluation/MSBuildEvaluator.cs

```diff
@@ -0,0 +1,201 @@
+                {
+                    _logger?.LogDebug("Project is not restored, exiting.");
+                    return MSBuildEvaluationResult.CreateNoRestore(projectPath);
+                }
+
+                //If target framework is set, no further evaluation is needed.
+                if (!string.IsNullOrWhiteSpace(targetFramework))
+                {
+                    _logger?.LogDebug("Project is SDK style, single TFM:{0}, evaluation succeeded.", targetFramework);
+                    return SDKStyleEvaluationResult.CreateSuccess(projectPath, targetFramework, evaluatedProject);
+                }
+
+                //If target framework is not set, then presumably it is multi-target project.
+                //If there are no target frameworks, it is not expected.
+                if (targetFrameworks == null)
+                {
+                    _logger?.LogDebug("Project is SDK style, but does not specify the framework.");
+                    return MSBuildEvaluationResult.CreateFailure(projectPath, string.Format(LocalizableStrings.MSBuildEvaluator_Error_NoTargetFramework, projectPath));
+                }
+
+                //For multi-target project, we need to do additional evaluation for each target framework.
+                Dictionary<string, Project?> evaluatedTfmBasedProjects = new Dictionary<string, Project?>();
+                foreach (string tfm in targetFrameworks)
+                {
+                    _logger?.LogDebug("Evaluating project for target framework: {0}", tfm);
+                    evaluatedTfmBasedProjects[tfm] = RunEvaluate(projectPath, tfm);
+                }
+                _logger?.LogDebug("Project is SDK style, multi-target, evaluation succeeded.");
+                return MultiTargetEvaluationResult.CreateSuccess(projectPath, evaluatedProject, evaluatedTfmBasedProjects);
+
+            }
+            catch (Exception e)
+            {
+                _logger?.LogDebug("Unexpected error: {0}", e.Message);
+                return MSBuildEvaluationResult.CreateFailure(projectPath, e.Message);
+            }
+        }
+
+        private static Project RunEvaluate(string projectToLoad, string? tfm = null)
+        {
+            if (File.Exists(projectToLoad))
+            {
+                ProjectCollection projectCollection = new ProjectCollection();
+
+                if (!string.IsNullOrWhiteSpace(tfm))
+                {
+                    projectCollection.SetGlobalProperty("TargetFramework", tfm);
+                }
+
+                var project = new Project(
+                        projectToLoad,
+                        projectCollection.GlobalProperties,
+                        toolsVersion: "Current",
+                        subToolsetVersion: null,
+                        projectCollection,
+                        ProjectLoadSettings.IgnoreMissingImports | ProjectLoadSettings.IgnoreEmptyImports | ProjectLoadSettings.IgnoreInvalidImports);
```

Comment:
Can you comment why you want to allow evaluation on projects that have these sorts of errors? Is it that the evaluation you're doing is best-effort and it's preferable to get partial/incorrect information than fail with "cannot evaluate project"?

Reply:
We do only best effort here, also the evaluation should be fast vs complete. 
The result of evaluation is used for the following:
- determining if the template can be run in the following context (constraints) based on Project Capabilities
- determining properties values that will be used in template content

We consider the cost of the error as not very high:
- worst case scenario the user can create a template, which should not be allowed to and it fails to compile/build --> likely user will remove it or fix it manually then
- or the template content will be corrupted and consequent build fails --> the user may fix the issues manually if needed
- or the user will not see that template that is expected --> but they can always override it with `--force`

Therefore, we should not fail on missing imports or invalid imports, if this is the case rather restore/build should fail.

Reply:
Sounds good, and IMO worth putting a summary of that reasoning in a comment in the source, in case the relative costs change in the future.